src/visualizers/visualizerSpace.js

```diff
@@ -2,12 +2,11 @@
 /**
  * Created by Michael on 31/12/13.
  * modified by henryz00
- * original repo: https://github.com/michaelbromley/soundcloud-visualizer
+ * original repo: https://github.canalyserom/michaelbromley/soundcloud-visualizer
  */
 let visualizer;
 let audioSource;
-let analyser;
-let dataArray;
+let audioData;
 let container;
```

Comment:
Do these variables really need to be file scoped? Can they be limited to `PlayerAudioSource`?

Reply:
It should, however they are used across two classes tho, need to find a way to refactor

Reply:
I would recommend that you define suitable accessors (and mutators) that enforce required invariants / constraints for these variables, instead of just passing references to them around.

Reply:
Like creating an object that includes all the shared variables directly in this file?

Reply:
IF they are used together (and that their lifetime should be tied together), then yes, that would make sense enforce access invariants and constraints that way,